lib/AST/NameLookup.cpp

```diff
@@ -2845,6 +2845,11 @@ createOpaqueParameterGenericParams(GenericContext *genericContext, GenericParamL
   if (!value)
     return { };
 
+  // Consider storage signature opaque parameters for accessor declaration. 
```

Comment:
I wonder if we can refactor things a bit and move the logic into GenericParamListRequest::evaluate().

Take a look at how the parser handles parsed generic parameter lists of subscripts today: they're cloned at parse time when the AccessorDecl is constructed, in createAccessorFunc() in ParseDecl.cpp:

```
  auto *D = AccessorDecl::create(P->Context,
                                 /*FIXME FuncLoc=*/DeclLoc,
                                 AccessorKeywordLoc,
                                 Kind, storage,
                                 StaticLoc, StaticSpellingKind::None,
                                 asyncLoc.isValid(), asyncLoc,
                                 throwsLoc.isValid(), throwsLoc,
                                 (GenericParams
                                  ? GenericParams->clone(P->CurDeclContext)
                                  : nullptr),
                                 ValueArg, Type(),
                                 P->CurDeclContext);
```

Instead, we could probably do this in GenericParamListRequest::evaluate(), and clone subscriptDecl->getGenericParamList() there instead. This would simplify some of the parser logic that threads the GenericParamList through parseGetSet().

Moving semantic analysis like that out of the parser will help simplify the ASTGen pass that we'll use once we migrate to the new swift-syntax parser.

Reply:
It makes total sense @slavapestov! I'll give that a try

Reply:
Actually should this be in this PR, or could be a follow-up?

Reply:
If you don't mind, I think in this PR is best since it will involve undoing the above change anyway.

Reply:
No problem :)

Reply:
Agree with Slava here, an accessor should not have a parsed generic parameter list.